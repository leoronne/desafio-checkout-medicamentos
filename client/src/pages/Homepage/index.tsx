/* eslint-disable react/no-array-index-key */
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { CircularProgress } from '@material-ui/core';

import { useLanguage } from '../../hooks';

import { ButtonOutlined } from '../../styles/MaterialUI';
import { Container, Content, Form, MedicineCard, MedicineIcon } from './styles';

const Homepage: React.FC = () => {
  const { language } = useLanguage();
  const { t } = useTranslation();
  const date = new Date();

  const [loading, setLoading] = useState(false);
  const [medicines, setMedicnes] = useState([
    {
      name: 'Ácido zoledrônico 4mg',
      price: 10.86,
      quantity: 1,
    },
    {
      name: 'Água para injeção 1mL',
      price: 30.06,
      quantity: 1,
    },
    {
      name: 'Betaserc 16mg',
      price: 8.46,
      quantity: 1,
    },
    {
      name: 'Bromazepam 3mg',
      price: 0.46,
      quantity: 1,
    },
  ]);

  return (
    <Container>
      <Content>
        <Form>
          <header>
            <h1>{t('homepage-title-1')}</h1>

            <p>
              {`${t('homepage-title-2')} ${new Intl.DateTimeFormat(language === 'pt' ? 'pt-BR' : 'en-US', {
                weekday: 'long',
                year: 'numeric',
                month: 'long',
                day: 'numeric',
              }).format(date)}.`}
            </p>
          </header>

          <main>
            {medicines.map((medicine, index) => (
              <MedicineCard key={index}>
                <div className="medicine-logo">
                  <MedicineIcon />
                </div>
                <div className="medicine-info">
                  <p>{`${t('quantity')}: ${medicine.quantity}`}</p>
                  <h2>{medicine.name}</h2>
                </div>
              </MedicineCard>
            ))}
          </main>

          <footer>
            <p className="footer-text">{t('homepage-footer-text-1')}</p>
            <ButtonOutlined type={!loading ? 'submit' : 'button'} disabled={loading} data-testid="submit-button">
              {loading ? (
                <div>
                  <CircularProgress size={15} style={{ color: '#ccc' }} />
                </div>
              ) : (
                <p>{t('find-phamarcy')}</p>
              )}
              <span>{t('sponsor')}</span>
            </ButtonOutlined>
          </footer>
        </Form>
      </Content>
    </Container>
  );
};

export default Homepage;
