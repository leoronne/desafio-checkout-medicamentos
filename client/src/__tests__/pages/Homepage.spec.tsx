import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import notistack from 'notistack';

// import { pharmacyResponse } from '../../__mocks__';

import { Homepage } from '../../pages';

const mockedHistoryPush = jest.fn();
const mockedloadData = jest.fn();
const enqueueSnackbar = jest.fn();
const closeSnackbar = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useHistory: () => ({
      push: mockedHistoryPush,
    }),
    Link: ({ children }: { children: React.ReactNode }) => children,
  };
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

jest.mock('notistack', () => ({
  useSnackbar: jest.fn(),
}));

jest.mock('../../hooks', () => {
  return {
    useLanguage: () => ({
      language: 'en',
    }),
  };
});

describe('Homepage', () => {
  beforeEach(() => {
    mockedHistoryPush.mockClear();
    mockedloadData.mockClear();

    jest.spyOn(notistack, 'useSnackbar').mockImplementation(() => {
      return { enqueueSnackbar, closeSnackbar };
    });
  });

  const initialState = { output: 10 };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('should render', async () => {
    await act(async () => {
      const { asFragment } = render(
        <Provider store={store}>
          <Homepage />
        </Provider>
      );

      const html = asFragment();

      expect(html).toMatchSnapshot();
    });
  });
});
