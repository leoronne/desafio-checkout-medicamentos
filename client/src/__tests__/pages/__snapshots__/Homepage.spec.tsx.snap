// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz QGLgt"
  >
    <div
      class="sc-dlfnbm jwCXoh"
    >
      <form
        class="sc-hKgILt hWhjlt"
      >
        <header>
          <h1>
            homepage-title-1
          </h1>
          <p>
            homepage-title-2 Friday, November 13, 2020.
          </p>
        </header>
        <main>
          <div
            class="sc-eCssSg hZNvkz"
          >
            <div
              class="medicine-logo"
            >
              <svg
                class="sc-bdfBwQ briHBD"
              >
                medicine.svg
              </svg>
            </div>
            <div
              class="medicine-info"
            >
              <p>
                quantity: 1
              </p>
              <h2>
                Ácido zoledrônico 4mg
              </h2>
            </div>
          </div>
          <div
            class="sc-eCssSg hZNvkz"
          >
            <div
              class="medicine-logo"
            >
              <svg
                class="sc-bdfBwQ briHBD"
              >
                medicine.svg
              </svg>
            </div>
            <div
              class="medicine-info"
            >
              <p>
                quantity: 1
              </p>
              <h2>
                Água para injeção 1mL
              </h2>
            </div>
          </div>
          <div
            class="sc-eCssSg hZNvkz"
          >
            <div
              class="medicine-logo"
            >
              <svg
                class="sc-bdfBwQ briHBD"
              >
                medicine.svg
              </svg>
            </div>
            <div
              class="medicine-info"
            >
              <p>
                quantity: 1
              </p>
              <h2>
                Betaserc 16mg
              </h2>
            </div>
          </div>
          <div
            class="sc-eCssSg hZNvkz"
          >
            <div
              class="medicine-logo"
            >
              <svg
                class="sc-bdfBwQ briHBD"
              >
                medicine.svg
              </svg>
            </div>
            <div
              class="medicine-info"
            >
              <p>
                quantity: 1
              </p>
              <h2>
                Bromazepam 3mg
              </h2>
            </div>
          </div>
        </main>
        <footer>
          <p
            class="footer-text"
          >
            homepage-footer-text-1
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-text"
            data-testid="submit-button"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <p>
                find-phamarcy
              </p>
              <span>
                sponsor
              </span>
            </span>
          </button>
        </footer>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
